tests/ui/projection/impl-projection.rs

```diff
@@ -0,0 +1,23 @@
+// check-pass
+
+#![crate_type = "lib"]
+
+pub trait Identity {
+    type Identity: ?Sized;
+}
+
+impl<T: ?Sized> Identity for T {
+    type Identity = Self;
+}
+
+pub struct S;
+
+impl <S as Identity>::Identity {
```

Comment:
Sorry, *this* is what I wanted to mention. I'm not sure if we *want* to be able to normalize impl headers when considering inherent impls.

Reply:
That's also something I wondered but after talking with @oli-obk, they seemed to think it was ok so I made this PR. I think it'll require an FCP in any case so you can agree with your team whether or not it's something you want.

Reply:
My words were "why should we not allow it", I don't know the answer to that. Mostly I don't know the use case for it

Reply:
Sorry for the confusion. Well in any case, I'll let your team discuss about it.

Reply:
I remember having written code exactly like that before and then being surprised that it didn't compile. I can't remember where that was and I must've found a different approach, but it seems like something that is consistent with how other parts of the language works. And it's kind of surprising/unexpected that it doesn't compile.